src/coreclr/jit/instr.cpp

```diff
@@ -176,13 +176,18 @@ const char* CodeGen::genSizeStr(emitAttr attr)
         "xmmword ptr ",
         nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr,
         nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr,
-        "ymmword ptr"
+        "ymmword ptr",
+        nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr,
+        nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr,
+        nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr,
+        nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr,
+        "zmmword ptr"
```

Comment:
Similar here as to the earlier case. This really seems like something we could compress down to 7 entries and use `genLog2`. Unlike the earlier case we can't remove the lookup table entirely, but we can save some space by having a 1 cycle penalty on a debug only lookup.

Reply:
We will still have to handle 0 if I understand this correctly
```suggestion
        const char* CodeGen::genSizeStr(emitAttr attr)
{
    // clang-format off
    static
    const char * const sizes[] =
    {
        "",
        "byte  ptr ",
        "word  ptr ",
        "dword ptr ",
        "qword ptr ",
        "xmmword ptr ",
        "ymmword ptr",
        "zmmword ptr"
    };
    // clang-format on

    unsigned size = EA_SIZE(attr);

    assert(size == 0 || size == 1 || size == 2 || size == 4 || size == 8 || size == 16 || size == 32 || size == 64);

    if (EA_ATTR(size) == attr)
    {
        return sizes[ size > 0 ? genLog2(size) + 1 : size];
    }
    else if (attr == EA_GCREF)
    {
        return "gword ptr ";
    }
```

Reply:
I think that's fine. We could likewise just have it as `return (size > 0) ? sizes[genLog2(size)] : "";`